File: 1464-maximum-product-of-two-elements-in-an-array/1464-maximum-product-of-two-elements-in-an-array.cpp
class Solution {
public:
    int maxProduct(vector<int>& nums) {
        int maxi = 0, secMaxi = 0;
        
        for(int i=0; i<nums.size(); i++) {
            if(nums[i]>maxi) {
                secMaxi = maxi;
                maxi = nums[i];
            } else {
                secMaxi = max(secMaxi, nums[i]);
            }
        }
        return (maxi-1)*(secMaxi-1);
    }
};

File: 1464-maximum-product-of-two-elements-in-an-array/1464-maximum-product-of-two-elements-in-an-array.go
func maxProduct(nums []int) int {
    maxi, secMaxi := 0, 0
    
    for _, num := range nums {
        if num>maxi {
            secMaxi = maxi
            maxi = num
        } else {
            secMaxi = max(secMaxi, num)
        }
    }
    return (maxi-1)*(secMaxi-1)
}